internal/appmain/appmain.go

```diff
@@ -33,26 +39,48 @@ var (
 
 // RunApplication starts and runs the given application forever.  For use in
 // main functions to run the full application.
-func RunApplication(serverName string, bindService Bind) {
-	StartApplication(serverName, bindService, config.Read)
+func RunApplication(serviceName string, bindService Bind) {
+	c := make(chan os.Signal, 1)
+	// SIGTERM is signaled by k8s when it wants a pod to stop.
+	signal.Notify(c, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
https://golang.org/pkg/os/signal/
> "The signals SIGKILL and SIGSTOP may not be caught by a program, and therefore cannot be affected by this package."

It will already have received a sigterm and started shutdown anyways.